src/test/java/net/masterthought/cucumber/ReportParserTest.java

```diff
@@ -88,4 +90,145 @@ public void parseJsonResults_OnNoExistingFile_ThrowsException() {
 
         reportParser.parseJsonFiles(jsonReports);
     }
+
+    @Test
+    public void parsePropertyFiles_EmptyProperties() {
+
+        // given
+        initWithProperties(EMPTY_PROPERTIES);
+        ReportParser reportParser = new ReportParser(configuration);
+
+        // when
+        reportParser.parsePropertiesFiles(propertyReports);
+
+        // then
+        assertThat(configuration.getClassifications()).isEmpty();
+    }
+
+    @Test
+    public void parsePropertyFiles_Populate_One_File() {
+
+        // given
+        initWithProperties(SAMPLE_ONE_PROPERTIES);
+        ReportParser reportParser = new ReportParser(configuration);
+
+        // when
+        reportParser.parsePropertiesFiles(propertyReports);
+
+        // then
+        assertThat(configuration.getClassifications()).hasSize(5);
+    }
+
+    @Test
+    public void parsePropertyFiles_Populate_Two_Files() {
+
+        // given
+        initWithProperties(SAMPLE_ONE_PROPERTIES, SAMPLE_TWO_PROPERTIES);
+        ReportParser reportParser = new ReportParser(configuration);
+
+        // when
+        reportParser.parsePropertiesFiles(propertyReports);
+
+        // then
+        assertThat(configuration.getClassifications()).hasSize(8);
+    }
+
+    @Test
+    public void parsePropertyFiles_Populate_Check_Contents_Integrity() {
+
+        // given
+        initWithProperties(SAMPLE_TWO_PROPERTIES);
+        ReportParser reportParser = new ReportParser(configuration);
+
+        // when
+        reportParser.parsePropertiesFiles(propertyReports);
+
+        List<Map.Entry<String, String>> classifications = configuration.getClassifications();
+
+        // then
+        assertThat(classifications).hasSize(3);
+        assertThat(classifications).containsExactly(entry("NodeJsVersion","8.5.0"),entry("Proxy", "http=//172.22.240.68:18717"),entry("NpmVersion", "5.3.0"));
+    }
+
+    @Test
+    public void parsePropertyFiles_Populate_Check_Contents_Order() {
+
+        // given
+        initWithProperties(SAMPLE_TWO_PROPERTIES);
+        ReportParser reportParser = new ReportParser(configuration);
+
+        // when
+        reportParser.parsePropertiesFiles(propertyReports);
+
+        List<Map.Entry<String, String>> classifications = configuration.getClassifications();
+
+        // then
+        assertThat(classifications).hasSize(3);
+        assertThat(classifications.get(0).getKey()).isEqualTo("NodeJsVersion");
+        assertThat(classifications.get(1).getKey()).isEqualTo("Proxy");
+        assertThat(classifications.get(2).getKey()).isEqualTo("NpmVersion");
+
+    }
+
+    @Test
+    public void parsePropertyFiles_Populate_Check_Duplicates() {
+
+        // given
+        initWithProperties(DUPLICATE_PROPERTIES);
+        ReportParser reportParser = new ReportParser(configuration);
+
+        // when
+        reportParser.parsePropertiesFiles(propertyReports);
+
+        List<Map.Entry<String, String>> classifications = configuration.getClassifications();
```

Comment:
again, what is `// when` and what is `// then` ?

Reply:
I'll make this a when as we are doing an action...ok with you?

Reply:
I like to have only one instruction in "// when" section

Reply:
ok fair enough I will be submitting a new PR with all the changes you requested in the next hour or so